### PR TITLE
refactor: ThemePage

### PR DESCRIPTION
**Pull request by edalex-yinzi**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [ ] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change
Convert ThemePage to functional component.
Create a story for ThemePage.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by edalex-yinzi** [comment]
Found two conflict variables `LogoSettings`...

File: react-front-end/tsrc/theme/ThemePage.tsx [hunk 1]
```diff
@@ -285,110 +296,108 @@ class ThemePage extends React.Component<
...
-      >
-        <this.ColorSchemeSettings />
-        <this.LogoSettings />
-      </SettingPageTemplate>
-    );
-  }
-}
+  return (
+    <SettingPageTemplate
+      onSave={() => saveChanges()}
```
Comment (edalex-ian) [inline react-front-end/tsrc/theme/ThemePage.tsx:389]:
```suggestion
      onSave={saveChanges}
```

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Looking good @edalex-yinzi . Mostly just tidy up and making things consistent with existing codebase.

BTY.

File: react-front-end/tsrc/theme/ThemePage.tsx [hunk 2]
```diff
@@ -82,149 +84,160 @@ interface ThemeColors {
...
+      background: "#fafafa",
+      paper: "#ffffff",
+      menuText: "#000000",
+      menu: "#ffffff",
+      menuIcon: "#000000",
+      primaryText: "#000000",
+      secondaryText: "#444444",
+    };
+    setThemeColors(defaultThemeColors);
+    setIsChangesUnsaved(true);
```
[the remark this answers is not in the document]
Comment (edalex-yinzi) [reply react-front-end/tsrc/theme/ThemePage.tsx:166]: It's the issue I mentioned in slack. I will delete this.

File: react-front-end/tsrc/theme/ThemePage.tsx [hunk 3]
```diff
@@ -82,149 +84,160 @@ interface ThemeColors {
...
-      secondaryText: themeSettings.menuTextColor,
-    });
+  const setColorPickerDefaults = () => {
+    pipe(mapThemeSettings2ThemeColors(), setThemeColors);
   };
 
-  reload = () => {
+  const reload = () => {
     window.location.reload();
   };
```
[the remark this answers is not in the document]
Comment (edalex-yinzi) [reply react-front-end/tsrc/theme/ThemePage.tsx]: Error said "Uncaught TypeError: Illegal invocation". Need bind context window. So I will only remove the `{}`.

File: react-front-end/tsrc/theme/ThemePage.tsx [hunk 4]
```diff
@@ -82,149 +84,149 @@ interface ThemeColors {
...
-    menu: themeSettings.menuItemColor,
-    menuText: themeSettings.menuItemTextColor,
-    menuIcon: themeSettings.menuItemIconColor,
-    primaryText: themeSettings.primaryTextColor,
-    secondaryText: themeSettings.menuTextColor,
+export const ThemePage = ({
+  updateTemplate,
+  classes,
+}: ThemePageProps & WithStyles<typeof styles>) => {
+  const mapSettingsToColors = (settings: IThemeSettings): ThemeColors => ({
```
Comment (edalex-ian) [inline react-front-end/tsrc/theme/ThemePage.tsx:101]: No action required on this, just FYI:

Aaaahhhh, this now highlights potentially why you were prefixing all the interfaces with `I`. Due to this (IThemeSettings) I went and had a closer look at the file it's in and that highlighted extra things to be done as part of the broader refactor you're doing here. I'll capture the detail in the internal Jira ticket.

But just quickly highlighting, in the work your doing to add the Theme endpoints to oeq-ts-rest-api I'd expect `IThemeSettings` to become `ThemeSettings` and live in something like `OEQ.Theme`.

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Looking good @edalex-yinzi, just one minor change to convert a `function` to an arrow function.

**Review by edalex-ian — APPROVED** [review]
Good stuff. Thanks @edalex-yinzi

**Pull request merged**